crates/profiling/src/vtune_linux.rs

```diff
@@ -132,13 +130,17 @@ impl State {
         functions: &PrimaryMap<DefinedFuncIndex, *mut [VMFunctionBody]>,
         _dbg_image: Option<&[u8]>,
     ) -> () {
+        // Global counter for module ids.
+        static MODULE_ID: atomic::AtomicUsize = atomic::AtomicUsize::new(0);
+
         for (idx, func) in functions.iter() {
             let (addr, len) = unsafe { ((**func).as_ptr() as *const u8, (**func).len()) };
             let default_filename = "wasm_file";
             let default_module_name = String::from("wasm_module");
             let module_name = module.name.as_ref().unwrap_or(&default_module_name);
             let method_name = super::debug_name(module, idx);
-            let method_id = self.get_method_id(module.id, idx);
+            let global_module_id = MODULE_ID.fetch_add(1, atomic::Ordering::SeqCst);
```

Comment:
I think this is allocating a new module id for each function, which I don't think is what happened before? (I'm not familiar with VTune so I'm not entirely certain)

Reply:
oh my, how embarrassing. Thanks :)